test_data/pil/permutation_via_challenges.pil

```diff
@@ -0,0 +1,23 @@
+let N: int = 8;
+namespace std::prover(N);
+    let challenge = [];
+    enum Query {
+        Hint(int)
+    }
+
+namespace Main(N);
+    col fixed first = [1] + [0]*;
+
+    // Two witness columns, claimed to be permutations of one another
+    col witness a(i) query std::prover::Query::Hint(i);
+    col witness b(i) query std::prover::Query::Hint(7 - i);
+
+    col witness stage(1) z;
+    let beta: expr = std::prover::challenge(0, 12345);
+
+    // First and last z needs to be 1
```

Comment:
I don't see the case about the last z.

Reply:
Because of wrapping it is the same. I added a comment to the file!

Reply:
For me, the last `z` is `z[N-1]`, that's what I meant.

Reply:
Conceptually, z is a vector of N+1 elements, starting and ending with 1. But because of wrapping, z[0] == z[N], so they are constrained to be equal anyway and we only need one constraint.
Actually, I think that is not even needed, as long as the accumulator starts and ends with the same value, it should be fine? Witgen would need a hint though.